### PR TITLE
Reduce Image library memory usage

### PR DESCRIPTION
**Pull request by Atlante45**
This PR changes the Image library to transfer ownership of the memory it uses down to the sub-
functions.
This lets those functions clear the memory after use and avoids it having multiple copies of the data laying around unnecessarily.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5365/HighFidelity-Beta-PR11920-cb743acecadd78dc7211709e726a5a20743333c8.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5365/HighFidelity-Beta-PR11920-cb743acecadd78dc7211709e726a5a20743333c8.dmg

File: libraries/image/src/image/Image.cpp [hunk 1]
```diff
@@ -238,33 +238,43 @@ uint32 packR11G11B10F(const glm::vec3& color) {
...
         newImageReader.setDecideFormatFromContent(true);
-        buffer.setData(content);
+        buffer.setData(localCopy);
         newImageReader.setDevice(&buffer);
 
         if (newImageReader.canRead()) {
             qCWarning(imagelogging) << "Image file" << filename.c_str() << "has extension" << filenameExtension.c_str()
-                                    << "but is actually a" << qPrintable(newImageReader.format()) << "(recovering)";
-            image = newImageReader.read();
+            << "but is actually a" << qPrintable(newImageReader.format()) << "(recovering)";
```
Comment (birarda) [inline libraries/image/src/image/Image.cpp]: Xcode changed the indentation of this on you

**Comment by Atlante45** [comment]
[Test Plan](https://highfidelity.testrail.net/index.php?/suites/view/58&group_by=cases:section_id&group_order=asc)
Only test the `Asset Server - Auto-Baking Tool` section.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5488/HighFidelity-Beta-PR11920-d6ac0dc3c2990767a64f5a72d78d953857c1ba99.dmg

**Comment by Atlante45** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5499/HighFidelity-Beta-PR11920-d6ac0dc3c2990767a64f5a72d78d953857c1ba99.dmg

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5502/HighFidelity-Beta-PR11920-d6ac0dc3c2990767a64f5a72d78d953857c1ba99.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5502/HighFidelity-Beta-PR11920-d6ac0dc3c2990767a64f5a72d78d953857c1ba99.dmg

**Comment by Nex-Pro** [comment]
Testing win10 and Vive

**Comment by Nex-Pro** [comment]
### FAIL

See testrail here.

https://highfidelity.testrail.net/index.php?/runs/view/1434&group_by=cases:section_id&group_id=1046&group_order=asc

Felt it necessary to fail until test rail cases and assets for test are clarified.

**Comment by Atlante45** [comment]
@Nex-Pro I updated the test plan.

**Comment by Nex-Pro** [comment]
@Atlante45   This now meets the updated requirements.

Also I was able to confirm we had a compressed mesh to test with.

**Comment by Atlante45** [comment]
Thanks @Nex-Pro!

**Pull request merged**